src/lib/zkapps_examples/big_circuit/zkapps_big_circuit.ml

```diff
@@ -0,0 +1,29 @@
+open Signature_lib
+
+(** Constructs a circuit with [num_constraints] constraints. *)
+let rec big_circuit (num_constraints : int) =
+  let open Snark_params.Tick.Run in
+  let b = exists Field.typ ~compute:(fun () -> Field.Constant.one) in
+  Field.Assert.equal b (constant Field.typ Field.Constant.one) ;
```

Comment:
Doesn't `Plonk_constraint_system` do constant unification? If so, I think this will only generate 1 constraint.

Reply:
You might have better luck with `assert_r1cs`

Reply:
I just did this to make sure that the `exists` create an actual constraint. I don't think it would do it otherwise